File: src/main/java/com/soapdogg/lispInterpreter/determiner/ExpressionNodeDeterminer.java
package com.soapdogg.lispInterpreter.determiner;

import com.soapdogg.lispInterpreter.datamodels.ExpressionNode;
import com.soapdogg.lispInterpreter.datamodels.Node;
import lombok.AllArgsConstructor;

@AllArgsConstructor(staticName = "newInstance")
public class ExpressionNodeDeterminer {

    public boolean isExpressionNode(final Node node) {
        return node instanceof ExpressionNode;
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/determiner/ExpressionNodeDeterminer.kt
package com.soapdogg.lispInterpreter.determiner

import com.soapdogg.lispInterpreter.datamodels.ExpressionNode
import com.soapdogg.lispInterpreter.datamodels.Node

class ExpressionNodeDeterminer {
    fun isExpressionNode(node: Node): Boolean {
        return node is ExpressionNode
    }
}